### PR TITLE
Per file code generation

### LINKED ISSUE #126: Separate output file for each source
I think it would be nice that `Sourcery` will create separate output files for each source wenn the `output` parameter is a folder.

### Example

```
sourcery source/ templates/ output/

├─ source
│   ├── Person.swift
│   └── Car.swift
├─ templates
│   ├── Coding.stencil
│   └── Description.stencil
└─ output
    ├── Person+Coding.swift
    ├── Person+Description.swift
    ├── Car+Coding.swift
    └── Car+Description.swift
```

### PR DESCRIPTION
**Pull request by ilyapuchka**
Resolves #126

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -522,12 +523,38 @@ If you want to attribute multiple items with same attributes, you can use sectio
...
+// sourcery:end
+
+}
 ```
 
 Sourcery will generate the template code and then perform replacement in your source file. Inlined generated code is not parsed to avoid chicken-egg problem.
 
+### Per file code generation
+
+Sourcery supporte generating code in a separate file per type, you just need to put `file` annotation in a template, e.g.
```
Comment (krzysztofzablocki) [inline README.md]: supports*

File: Sourcery/Parsing/FileParser.swift [hunk 2]
```diff
@@ -61,9 +61,9 @@ final class FileParser {
     ///
     /// - Returns: All types we could find.
     public func parse() -> FileParserResult {
-        let inline = InlineParser.parse(initialContents)
+        let inline = TemplateAnnotationsParser.parseAnnotations("inline", contents: initialContents)
```
Comment (krzysztofzablocki) [inline Sourcery/Parsing/FileParser.swift]: you should update var names as well in this case

Comment (ilyapuchka) [inline Sourcery/Parsing/FileParser.swift]: what vars do you men?

Comment (krzysztofzablocki) [inline Sourcery/Parsing/FileParser.swift]: ah maybe not, I wrote this before I saw how you were using template annotation parser

**Comment by maximkrouk** [comment]
So how can I generate some boilerplate in a file next to a source file?
I tried
```
{% for type in types.all %}
// sourcery:file:{{ file.path }}/{{ file.name }}+AutoCasePath
// sourcery:end
{% endfor %}
```
But it didn't work :c

**Pull request merged**